doc/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,143 @@
+# Troubleshooting
+
+Mountpoint for Amazon S3 is optimized for applications that need high read throughput to large objects, potentially from many clients at once,
+and to write new objects sequentially from a single client at a time.
+To achieve this, Mountpoint does not implement all the features of a POSIX file system that may affect compatibility with your application.
+
+This documentation enumerates some examples of what error messages may look like when trying to perform unsupported operations. Please also take a look at Mountpoint's [semantics documentation](../doc/SEMANTICS.md) for more more detailed information on the decisions and tradeoffs made.
+This troubleshooting page is based on release Mountpoint-s3 v1.4.0 .Versions earlier to that might not have same error logging.
+
+## Random Write
+
+Mountpoint supports writing to a file sequentially. Random writes, or 'out-of-order' writes, will return an error to FUSE, which may appear in applications with the error message "Invalid argument".
+For example, the following code seeks one byte into the file and writes a single byte using `dd`.
+
+```
+$ dd if=/dev/random of=out seek=1 count=1
+dd: writing to 'out': Invalid argument
+```
+
+In Mountpoint's logs, a warning message will be emitted similar to below:
+
+```
+WARN write{req=52 ino=49 fh=3 offset=512 length=512 name="out"}: 
+mountpoint_s3::fuse: write failed:upload error: out of order write NOT supported by Mountpoint, aborting the upload; expected offset 0 but got 512
+```
+
+## Writing to an existing File 
+
+Mountpoint support overwriting to existing file using mount option `--allow-overwrite` . 
+Trying to open an existing file for writing without `--allow-overwrite`  flag will fail with the error: `Operation not permitted`.
+For example, there is an pre-existing file 'existing-file.txt' in mounted directory. We try to overwrite the 'existing-file' as follow:
+
+```
+$ echo "Overwriting a file..." > existing-file.txt
+operation not permitted: existing-file.txt
+```
+
+Log entries for file overwriting looks like:
+
+```
+WARN setattr{req=4 ino=21 name="existing-file.txt"}: 
+mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified
+```
+
+### Unreleased
```

Comment:
this is hard to follow, may be put here only the final version and merge this page when the new release was created? probably, leaving a note like the following for prev versions, though:

> Note: in versions of MP before 1.5.0 the error will look like `mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified`

Reply:
I think we can merge this troubleshooting page for now and make the change once the new release is made. Because we don't know when are we going to do next patch release. If it is soon enough, I think we can wait.

Reply:
It might be helping users who build from source but also confusing for users that use official releases. I would not include it in the doc for now.

Reply:
In that case, I think we can keep troubleshooting page updated with #718 and once that PR gets released in something like v1.4.1, we can merge this PR (soon after or before??)
Also, how do we make sure that this PR is ready to be merged but not merge it till next release? Just approve it and wait? We might need to rebase (although unlikely) and be careful to not merge it before the release.

Reply:
Yeah, its frustrating to have PRs open for too long. You could put something like this to unblock the merge:
___

Log entries for file overwrites looks like one of the following depending on the Mountpoint version:
```
WARN setattr{req=4 ino=21 name="existing-file.txt"}: 
mountpoint_s3::fuse: setattr failed: inode error: inode 21 (full key "existing-file.txt") is a remote inode and its attributes cannot be modified
```
```
WARN setattr{req=11 ino=2 name="existing-file.txt"}: 
mountpoint_s3::fuse: setattr failed: file overwrite is disabled by default, you need to remount with --allow-overwrite flag and open the file in truncate mode (O_TRUNC) to overwrite it
```
___

It's just this `Unreleased` looks a little off here.